### PR TITLE
Save state of traces panel to localstorage

### LINKED ISSUE #54: Save showing/hiding state to localstorage
Same as #51, but for showing/hiding state

### PR DESCRIPTION
**Pull request by saskali**
fixes #54 

- Save tracing panel status for showing/hiding, traces and showing/hiding state to localstorage

### CODE REVIEW

**Comment by daiyi** [comment]
hej @Dexterminator! Nice to meet you (:

**Comment by saskali** [comment]
thanks for reviewing @Dexterminator! 

I misunderstood issue #54 and modified this PR so that the whole state of the trace panel is being saved to localstorage. This includes: 

- panel is shown or hidden
- traces that have been captured
- detail expansion or collapse

 @danielcompton what are your thoughts about this? anything that needs to be adjusted?

File: src/day8/re_frame/trace.cljs [hunk 1]
```diff
@@ -191,7 +192,15 @@
         filter-items               (r/atom (localstorage/get "filter-items" []))
         filter-type                (r/atom :contains)
         input-error                (r/atom false)
-        trace-detail-expansions    (r/atom {:show-all? false :overrides {}})]
+        trace-detail-expansions    (r/atom (localstorage/get "trace-detail-expansion" {:show-all? false :overrides {}}))]
+    (add-watch traces
+               :update-traces
+               (fn [_ _ _ new-state]
+                 (localstorage/save! "traces" new-state)))
```
Comment (danielcompton) [inline src/day8/re_frame/trace.cljs]: Don't save the traces, just the other settings.

**Comment by saskali** [comment]
ready to review!

**Comment by danielcompton** [comment]
It's ok, having re-frame would make this a lot cleaner.

**Pull request merged**